addon/components/frost-password.js

```diff
@@ -1,103 +1,82 @@
+  type (isRevealed) {
+    return isRevealed ? 'text' : 'password'
+  },
 
-  // ==========================================================================
-  // Events
-  // ==========================================================================
+  // == Events ================================================================
 
-  focusOut: Ember.on('focusOut', function () {
+  _focusOut: on('focusOut', function (event) {
+    // this._super(...arguments)
     this.set('isCapsOn', false)
   }),
 
-  keyDown: Ember.on('keyDown', function (e) {
+  _keyDown: Ember.on('keyDown', function (e) {
     var s = e || window.e
-    if (this.get('isCapsOn') === false && (s.which === 20 || s.keyCode === 20)) {
+    if (s.which === 20 || s.keyCode === 20) {
       this.set('isCapsOn', true)
     }
   }),
 
-  keyUp: Ember.on('keyUp', function (e) {
-    var s = e || window.e
-    if (this.get('isCapsOn') === true && (s.which === 20 || s.keyCode === 20)) {
-      this.set('isCapsOn', false)
-    }
-  }),
-
-  keyPressed: Ember.on('keyPress', function (e) {
-    var s = String.fromCharCode(e.which || e.keyCode)   // IE support
-    if ((s.toUpperCase() === s && s.toLowerCase() !== s && !e.shiftKey) ||
-      (s.toUpperCase() !== s && s.toLowerCase() === s && e.shiftKey)) { // caps is on
+  _keyPress: Ember.on('keyPress', function (event) {
+    const c = String.fromCharCode(event.which || event.keyCode)
+    if ((c.toUpperCase() === c && c.toLowerCase() !== c && !event.shiftKey) ||
+      (c.toUpperCase() !== c && c.toLowerCase() === c && event.shiftKey)) {
       this.set('isCapsOn', true)
     }
   }),
 
-  _onFocus: Ember.on('focusIn', function (e) {
-    // If an onFocus handler is defined, call it
-    if (this.attrs.onFocus) {
-      this.attrs.onFocus()
+  _keyUp: Ember.on('keyUp', function (e) {
+    var s = e || window.e
+    if (s.which === 20 || s.keyCode === 20) {
+      this.set('isCapsOn', false)
     }
   }),
 
-  // ==========================================================================
-  // Actions
-  // ==========================================================================
+  // == Actions ===============================================================
 
   actions: {
-    onBlur () {
-      const onBlur = this.get('onBlur')
-
-      if (onBlur) {
-        onBlur()
-      }
-    },
-
-    onInput (args) {
-      this.set('isRevealerVisible', args.value.length > 0)
-      if (_.isFunction(this.get('onInput'))) {
-        this.get('onInput')(args)
-      }
-    },
     toggleReveal () {
-      this.toggleProperty('revealed')
-      this.set('type', this.get('revealed') ? 'text' : 'password')
-      this.set('revealIcon', this.get('revealed') ? 'hide' : 'show')
+      this.toggleProperty('isRevealed')
+      const $input = this.$('input').get(0)
+      $input.focus()
+      // Move the cursor to the end of the input
+      run.next(this, function () {
+        $input.selectionStart = $input.selectionEnd = 10000
```

Comment:
Since `$input` is referenced in a `run.next()` call I think it is possible for the DOM to already be destroyed and thus we probably need a sanity check here.

```js
run.next(this, function () {
  if (document.contains($input[0]) {
    $input.selectionStart = $input.selectionEnd = 10000
  }
})
```